### ISSUE TITLE
jax.nn.sigmoid gradient returns zero for large positive float64 input

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#40385]
### Description

`jax.nn.sigmoid` appears to return an incorrect zero first-order derivative for a finite `float64` input.

For sigmoid, the derivative can be written as:

```
sigmoid'(x) = exp(-x) / (1 + exp(-x))**2
```

At `x = 37.42994775023705`, the high-precision reference rounded to `float64` is approximately `5.551115123125775e-17`, which is finite, normal, and nonzero. However, JAX returns `0.0`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

x = 37.42994775023705
expected = 5.551115123125775e-17

actual = float(jax.grad(lambda z: jnp.sum(jax.nn.sigmoid(z)))(
    jnp.asarray(x, dtype=jnp.float64)
))

print("jax:", actual)
print("expected:", expected)

if not (math.isfinite(actual) and abs(actual - expected) <= 1e-6 * abs(expected)):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: 0.0
expected: 5.551115123125775e-17
```

### Expected result

```
5.551115123125775e-17
```

The expected derivative is finite, normal, nonzero, and representable in `float64`. JAX instead returns zero, suggesting that the autodiff path loses a small but representable derivative signal.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

**Comment by jakevdp** [comment]
This falls under #38651 – issues like this (typically discovered via AI fuzzing of extreme input values) are low priority for the JAX team.
